tv/src/main/java/butter/droid/tv/ui/trailer/TVTrailerPlayerActivity.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * This file is part of Butter.
+ *
+ * Butter is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Butter is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Butter. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package butter.droid.tv.ui.trailer;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.support.v4.app.FragmentManager;
+import android.util.Log;
+import android.view.MenuItem;
+import butter.droid.base.manager.network.NetworkManager;
+import butter.droid.base.manager.youtube.YouTubeManager;
+import butter.droid.base.providers.media.models.Media;
+import butter.droid.base.torrent.StreamInfo;
+import butter.droid.base.torrent.TorrentService;
+import butter.droid.tv.R;
+import butter.droid.tv.TVButterApplication;
+import butter.droid.tv.activities.base.TVBaseActivity;
+import butter.droid.tv.fragments.TVPlaybackOverlayFragment;
+import butter.droid.tv.fragments.TVVideoPlayerFragment;
+import java.lang.ref.WeakReference;
+import java.net.URLDecoder;
+import javax.inject.Inject;
+import timber.log.Timber;
+
+public class TVTrailerPlayerActivity extends TVBaseActivity implements TVVideoPlayerFragment.Callback, TVTrailerPlayerView {
```

Comment:
there is a lot of duplicated code here. You should have abstract `BaseTrailerPlayerActivity` and then just differences in mobile and tv version. `BaseStreamLoadingFragment` is done in this way for example.